In [2]:
%pip install pybaseball -q

     |████████████████████████████████| 415 kB 5.2 MB/s 
     |████████████████████████████████| 291 kB 48.6 MB/s 
     |████████████████████████████████| 856 kB 38.2 MB/s 


In [3]:
import pybaseball as pyb
from pybaseball import statcast
from pybaseball import statcast_pitcher, playerid_lookup
from pybaseball import pitching_stats_bref

In [4]:
import pandas as pd
import numpy as np

In [11]:
cole_id = playerid_lookup('cole','gerrit')['key_mlbam'][0]
cole_id

543037

In [28]:
cole = statcast_pitcher('2021-04-01', '2021-10-03', player_id = cole_id)
cole.head(5)

Gathering Player Data


pitch_type   game_date  release_speed  release_pos_x  release_pos_z  \
0         CH  2021-09-29           90.3          -1.83           5.55   
1         CH  2021-09-29           89.4          -1.78           5.57   
2         FF  2021-09-29           97.4          -1.86           5.71   
3         SL  2021-09-29           87.5          -1.73           5.75   
4         SL  2021-09-29           87.9          -1.66           5.85   

    player_name  batter  pitcher     events    description  ...  fld_score  \
0  Cole, Gerrit  624512   543037  field_out  hit_into_play  ...          3   
1  Cole, Gerrit  624512   543037        NaN       foul_tip  ...          3   
2  Cole, Gerrit  624512   543037        NaN           ball  ...          3   
3  Cole, Gerrit  669289   543037     single  hit_into_play  ...          3   
4  Cole, Gerrit  669289   543037        NaN  called_strike  ...          3   

   post_away_score  post_home_score  post_bat_score  post_fld_score  \
0                3                5               5               3   
1                3                5               5               3   
2                3                5               5               3   
3                3                5               5               3   
4                3                5               5               3   

  if_fielding_alignment of_fielding_alignment spin_axis delta_home_win_exp  \
0              Standard              Standard     243.0             -0.014   
1              Standard              Standard     240.0              0.000   
2              Standard              Standard     224.0              0.000   
3              Standard              Standard      75.0              0.007   
4              Standard              Standard     125.0              0.000   

  delta_run_exp  
0        -0.215  
1        -0.048  
2         0.036  
3         0.139  
4        -0.018  

[5 rows x 92 columns]

In [21]:
cole_hh = cole[cole['launch_speed'] >= 95.0]

## Gerrit Cole 2021 WHHIP

In [24]:
hits = ['single', 'double','triple','home_run']

In [25]:
cole_hh_allowed = cole_hh[cole_hh['events'].isin(hits)]
hard_hits_allowed = cole_hh_allowed.shape[0]
hard_hits_allowed

81

Gerrit Cole BB: 41

Gerrit Cole IP: 181.1

In [26]:
cole_WHHIP = round((41 + hard_hits_allowed)/(181.33), 3)

In [27]:
cole_WHHIP

0.673

2021 Gerrit Cole's Hard-Hit WHIP (WHHIP): 0.673

---



## Gerrit Cole 2021 WHHIP.v2

In [29]:
# One Line of Code That is Changed
cole_hard_hits = cole_hh[cole_hh['description'] == "hit_into_play"]

In [30]:
all_hard_hits = cole_hard_hits.shape[0]
all_hard_hits

170

Gerrit Cole BB: 41

Gerrit Cole IP: 181.1

In [32]:
WHHIP_v2_cole = (41 + all_hard_hits)/(181.33)
print(round(WHHIP_v2_cole, 3))

1.164


Gerrit Cole 2021 WHHIP.v2: 1.164